src/main/java/com/github/kagkarlsson/scheduler/SchedulerClient.java

```diff
@@ -25,11 +26,23 @@
 import com.github.kagkarlsson.scheduler.task.TaskInstance;
 
 public interface SchedulerClient {
-	
-	void scheduleForExecution(Instant exeecutionTime, TaskInstance taskInstance);
-	
+
+	/**
+	 * This method has been renamed and will be removed in a future version.
+	 *
+	 * @deprecated use {@link #schedule(Instant, TaskInstance)} instead.
+	 */
+	@Deprecated
+	void scheduleForExecution(Instant executionTime, TaskInstance taskInstance);
+
+	void schedule(Instant executionTime, TaskInstance taskInstance);
+
+	void reschedule(String taskName, String instanceId, Instant newExecutionTime);
```

Comment:
You're right. It seems to be quite intrusive so, as you say, might be better to do in another PR/version.